cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -43,6 +50,7 @@ type SimulateTransactionResponse struct {
 	Results         []SimulateHostFunctionResult `json:"results,omitempty"`         // an array of the individual host function call results
 	Cost            SimulateTransactionCost      `json:"cost,omitempty"`            // the effective cpu and memory cost of the invoked transaction execution.
 	RestorePreamble *RestorePreamble             `json:"restorePreamble,omitempty"` // If present, it indicates that a prior RestoreFootprint is required
+	StateDiff       []LedgerEntryDiff            `json:"stateDiff,omitempty"`       // If present, it indicates how the state (ledger entries) will change as a result of the transaction execution.
```

Comment:
The term diff doesn't feel quite right, at the least what's included in the API doesn't really live up to the term. A diff is not actually returned, but rather the before and after snapshots are returned and it's up to the client to determine the diff if they need a diff.

Reply:
> We could use the term `stateChanges` instead? Other ideas?

Makes sense, I will change it.